print newest deps

cdzwm opened the pull request:
When conflict deps found, print the depency of newest version.

Commit:
Sugguest newest depency

Commit:
Fix a error

Commit:
Change print-newest-dep to private

Commit:
remove :eval-in :leiningen from project.clj

technomancy commented:
Thanks for this patch.

I think overall it's good to surface this information, but the way it's presented in this implementation is not particularly clear. Simply saying `Suggest: ...` is confusing because it's presented alongside the suggestion to use `:exclusions` and it's not obvious how the two different suggestions are related. I think this needs some more thought put into it, but we could include it if the information is presented in a less confusing way.

There's also a couple technical issues with it; for one it removes `:eval-in :leiningen` which is pretty important, so that needs to be changed.

Also it appears to bring in a new dependency, which causes a lot of extra work for downstream packagers. As far as I can tell, this is only being used for a single function, which duplicates logic already available from Aether: https://github.com/technomancy/leiningen/blob/master/leiningen-core/src/leiningen/core/pedantic.clj#L68 Let's use the existing implementation instead of bringing in something redundant.

Commit:
remove clj-version depency

Commit:
test use version

Commit:
test use node

Commit:
sort by node<

Commit:
merge remote changes

Commit:
clean code

Commit:
remove some waste code

Commit:
Print the newest version of the project confusing depencies.

cdzwm commented:
According to you suggestion, I make some modifications.

technomancy commented:
Sorry, I haven't gotten a chance to look at this yet, but I don't think it needs to be closed. Was that intentional?

cdzwm commented:
II'd like to think about this modification again. I think there are still some problems, but you can also take a look at it first and suggest some changes.

cdzwm committed:
Merge branch 'master' into upgrade-to-newest-deps

cucu committed:
pom.xml

cucu committed:
merge master

cucu committed:
change scm tag

cucu committed:
revert some format of code

cucu committed:
revert some format of code

cucu committed:
update scm tag

cdzwm commented:
Fix is completed.

cucu committed:
update scm tag

rlbdv committed:
Add "repl :headless :socket PATH" support

And match the configuration arrangments for the host, i.e. support
LEIN_REPL_SOCKET, :repl-options :socket, and an nrepl config :socket.

rlbdv committed:
Add a simple "repl :headless :socket" test

This relies on an unreleased version of nrepl (likely to be 1.0.0).
It also doesn't completely clean up after itself because if nothing
else, the (server ...) future will be blocked forever on a promise.

rlbdv committed:
Disable test-headless-socket until nrepl 1.0.0 is released

technomancy committed:
Add CI config for woodpecker.

technomancy committed:
Even simpler CI config.

technomancy committed:
Just use a normal debian image.

technomancy committed:
Add IRC notifications for failed builds.

technomancy committed:
Update all links to point to codeberg instead of github.

technomancy committed:
Move the "moved repository" notice to higher in the readme.

technomancy committed:
Codeberg doesn't use blob or tree in its URLs.

technomancy committed:
Do pedantic cycle detection based on strings, not dependency nodes.

Fixes https://github.com/technomancy/leiningen/issues/2798

This is caused by what seems like a bug in Aether's DependencyNode
implementation; most nodes will hash correctly such that

    (some #{node} parents)

works to do cycle detection, but certain ones in Netty's boringssl do
not hash correctly, so we have to convert them to strings before using
them for cycle detection purposes.

technomancy committed:
Hard-code names of built-in templates instead of querying classpath.

According to https://github.com/technomancy/leiningen/issues/2345 Java
9 broke the ability to query the classpath in the way we were doing
thru bultitude previously. Since Java 8 is in the minority now, we
will hardcode the template names instead of querying the classpath for
them.

This means that we can't show templates which come from plugins any
more, but it's better than showing nothing.

Fixes https://github.com/technomancy/leiningen/issues/2565

Fixes https://codeberg.org/leiningen/leiningen/issues/1

technomancy committed:
Merge branch 'domain-socket-repl' into main

technomancy committed:
Don't allow built-in templates to shadow remote templates.

We were stripping off the ns when we checked for built-in templates,
but now we use the full input for the built-in check.

Also add a new test for a template named default.

technomancy committed:
Update link to CI badge.

technomancy committed:
Update year in web page and remove tabs.

technomancy committed:
Recommend manual install in /usr/local/bin

technomancy committed:
Update changelog.

technomancy committed:
Release 2.9.9

technomancy committed:
Bump to version 2.9.10-SNAPSHOT

technomancy committed:
Update release steps.

technomancy committed:
Add test for dev-resources in uberjar.

technomancy committed:
Add hinting to xml/parse calls in uberjar and search.

The default implementation makes reflective calls here and triggers
warnings on post-java-8 versions:

WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by clojure.lang.InjectedInvoker/0x000000084029a840 (file:/home/phil/.m2/repository/org/clojure/clojure/1.11.1/clojure-1.11.1.jar) to method com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl.parse(org.xml.sax.InputSource,org.xml.sax.HandlerBase)
WARNING: Please consider reporting this to the maintainers of clojure.lang.InjectedInvoker/0x000000084029a840
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

technomancy committed:
Expand profiles before calculating which to remove in jar task.

technomancy committed:
Fix uberjar test to use input streams instead of relying on reflection.

technomancy committed:
Update bootclasspath deps listing.

technomancy committed:
Changelog bump; switch ci to jdk 17.

technomancy committed:
Release 2.9.10

technomancy committed:
bump version up to 2.9.11-SNAPSHOT

technomancy committed:
Relocation notice on web site.

christoph-frick committed:
Extract major if-branches into functions

Creating patches for tools like SDKMAN is quite hard, because huge
if-blocks need to change (no self-hosting, no up/downgrade).

Commit:
Merge pull request 'Extract major if-branches into functions' (#6) from cfrick/leiningen:extract-selfhosting-upgrade-shell-functions into main

Reviewed-on: https://codeberg.org/leiningen/leiningen/pulls/6

christoph-frick committed:
Move the check for the missing lein.jar out of run_from_checkout

In an "installed by maintainer" scenario the lein.jar will be in the
right place; this allows re-using `run_from_checkout` (at least for
SDKMAN).

christoph-frick committed:
Realign lein and lein-pkg scripts

This does minute movement of some items in `lein` and brings some
things, that never made it into `lein-pkg`.

technomancy committed:
Reindent

christoph-frick committed:
Use functions from lein in lein-pkg

This step makes both files quite equal and easier to maintain with a
"diff tool".  From lein the install and run-from-source branches where
removed and the now unused functions removed.

This also brings over the cygwin special treatment. Although there
currently is no leiningen package for cygwin, this makes maintaining
the two scripts easier.

Commit:
Merge pull request 'Realign lein and lein-pkg scripts' (#8) from cfrick/leiningen:realign-lein-and-lein-pkg into main

Reviewed-on: https://codeberg.org/leiningen/leiningen/pulls/8

technomancy committed:
Remove redundant profile merge for warning of implicit AOT.

technomancy committed:
Ensure default profile doesn't affect target-path.

Fixes https://codeberg.org/leiningen/leiningen/issues/9

technomancy committed:
Implement signing deploy artifacts using SSH keys, not just GPG.

technomancy committed:
Add documentation for signing deploy artifacts with SSH.

technomancy committed:
Add openssh-client to CI config so we can test SSH key signing.

technomancy committed:
The bin/notify script doesn't actually work in CI, of course.

technomancy committed:
Don't really need to do CI on openjdk 11 any more.

technomancy committed:
Manually set permissions of SSH signing key.

Git refuses to store the permissions as we set them.

technomancy committed:
Fix a bug with eval-in leiningen test exit codes.

technomancy committed:
Fix tests in the test task; this actually throws now.

technomancy committed:
Add an exclusion on jsoup as it's not actually used.

technomancy committed:
Bump wagon-http version and nrepl version.

technomancy committed:
Remove excluded profiles after the include-profiles have been expanded

We want to expand the profiles earlier on, but if we do this we can
lose the metadata required for pom-scoping. So instead we do it after
saving off the include-profiles-meta but before actually looping over.

I don't think this actually caused bugs because we were dissoc'ing the
profiles out of the profile map, but it would cause a spurious warning
that the profile in question was not found.

technomancy committed:
Suppress signing output in deploy tests.

technomancy committed:
Bump to 2.10.0

technomancy committed:
Release 2.10.0

technomancy committed:
Bump version to 2.10.1-SNAPSHOT.

technomancy committed:
Linking the wiki back to github since the codeberg wiki doesn't work right.

The whole point of the wiki is to be editable, yall. Come on.

technomancy committed:
Add 404 to web site, and clean up a few things.

We really don't need to keep IE compatibility, lol.

technomancy committed:
Add passthru to https://wiki.leiningen.org.

technomancy committed:
Update docs to point to wiki.leiningen.org.

technomancy committed:
Add warnings for buggy composite profiles.

lread committed:
tests: fix invalid pomegranate checksum option

Closes #15

technomancy committed:
Ensure top-level exceptions show their error message during test.

Previously we fixed this in 0b37859 but that had to be reverted
because it failed to propagate failures in :eval-in :leiningen.

See https://github.com/technomancy/leiningen/issues/1859

Commit:
Merge pull request 'tests: fix invalid pomegranate checksum option' (#16) from lread/leiningen:lread-test-checksum-value-fix into main

Reviewed-on: https://codeberg.org/leiningen/leiningen/pulls/16

technomancy committed:
Remove test for issue with older version of gpg.

We won't be making any further changes to this functionality anyway.

technomancy committed:
Suppress composite profile warning in tests.

technomancy committed:
Bring back old test :eval-in :leiningen behavior of returning exit code.

Move the throw back to the task itself for now; old way was better but
caused a bunch of test failures.

technomancy committed:
Pass the message on thru from abort to exit.

We're still double-printing in a really stupid way; still need to fix
up some bad tests to not depend on the bad behavior.

christoph-frick committed:
Fix link to lein.css (rel vs abs)

christoph-frick committed:
Fix typo with monospace fallback style for code blocks

christoph-frick committed:
Hide the line under the copyable anchors for the wiki

Commit:
Merge pull request 'Fix link for lein.css & some minor style changes' (#18) from cfrick/leiningen:wiki/fix-link into main

Reviewed-on: https://codeberg.org/leiningen/leiningen/pulls/18

technomancy committed:
Note about plagiarism in contributions.

christoph-frick committed:
Add sample section about :repl-options :history-file

... and how to disable the REPL history completely

Commit:
Merge pull request 'Add sample section about :repl-options :history-file' (#20) from cfrick/leiningen:add-repl-options-history-file into main

Reviewed-on: https://codeberg.org/leiningen/leiningen/pulls/20

technomancy committed:
Publish markdown documentation on leiningen.org.

We should link to this instead of the stuff in the git repo because it
will have better uptime.

technomancy committed:
Remove doctoc mention thingy; switch to pandoc-friendly language specifier

technomancy committed:
Allow badssl test to use a more general exception class.

technomancy committed:
Add support for XDG config directories.

Commit:
leiningen.repl: eliminate reflection in server-forms output

This commit does two things:

1. Add type hints to the REPL port files in the server-forms output

2. When an nREPL server is being started, either via the :start or
:headless subcommands, we bind *eval-print-dup* to true. This allows
type hints in the server-forms output to persist in the tempfile
generated by Leiningen.

The overall effect of these two changes is the elimination of reflection
when setting up a REPL.

technomancy committed:
Fix the fallback support for the old release plugin.

When we introduced the built-in release task, it shadowed an existing
"release" plugin, so we added support to prefer the plugin if it was
found.

We had a check in here to avoid an infinite loop of the actual release
task loading the actual release task forever, but it was checking
based on the filename, not the file contents.

technomancy committed:
Use XDG_CACHE_HOME instead of XDG_DATA_HOME.

technomancy committed:
Try to fix caching in circleci build.

technomancy committed:
And what about the codeberg CI; can we fix that too?

Commit:
leiningen.search: type hint query in search function

Removes one reflection warning when invoking the function

Commit:
leiningen.search: move response code checking to parse function

This ensures that a 400 and 505 response returned from Clojars (or
Maven Central) will give a meaningful error message to the user. In
particular, the user will know that their query syntax is unsupported.

In case of other errors, we fall back to providing the exception
message.

Commit:
introduce unit tests for leiningen.search/parse

This commit adds a new namespace, leiningen.test.search, where we set
up an HTTP server mocking Clojars error responses when invalid queries
are sent. This is done to test the behavior of leiningen.search/parse
when a 400 repsonse code is returned from the server. The overall
result is ensuring the previous commit improves error reporting in the
search task.

Commit:
Merge pull request 'Improved error reporting in search task' (#25) from rome-user/leiningen:search-patch into main

Reviewed-on: https://codeberg.org/leiningen/leiningen/pulls/25

technomancy committed:
Port CI config to JSON instead of YAML.

marcobiscaro2112 committed:
Upgrade clj-commons/pomegranate to 1.2.23

The new version has the latest fixes, including the removal of all reflective accesses [1].

This requires also upgrading org.apache.maven.wagon/wagon-http to 3.5.3 (which is the version of maven-wagon that pomegranate now uses).

Note that the new maven-wagon version does not depend on commons-io anymore, so we add that dependency explicitly (since we use org.apache.commons.io.output.TeeOutputStream).

Related to https://codeberg.org/leiningen/leiningen/issues/29

This marginally decreased the run time (~3% faster) for a pathological case.

Before:
```
$ lein --version
Leiningen 2.10.0 on Java 21.0.1 OpenJDK 64-Bit Server VM

$ hyperfine 'lein deps :tree'
Benchmark 1: lein deps :tree
  Time (mean ± σ):     144.215 s ±  2.752 s    [User: 164.353 s, System: 5.750 s]
  Range (min … max):   140.744 s … 149.894 s    10 runs
```

After:
```
$ lein-main --version
Leiningen 2.10.1-SNAPSHOT on Java 21.0.1 OpenJDK 64-Bit Server VM

$ hyperfine 'lein-main deps :tree'
Benchmark 1: lein-main deps :tree
  Time (mean ± σ):     140.863 s ±  3.346 s    [User: 160.611 s, System: 5.742 s]
  Range (min … max):   138.345 s … 148.004 s    10 runs
```

[1] https://github.com/clj-commons/pomegranate/pull/131

marcobiscaro2112 committed:
Avoid reflective access on lein deps :tree

Related to https://codeberg.org/leiningen/leiningen/issues/29

This significantly decreases the run time (~66% faster) and memory allocations (~68% less memory allocated) for a pathological case.

Baseline (commit d0b2a51):
```
$ hyperfine 'lein-main deps :tree'
Benchmark 1: lein-main deps :tree
  Time (mean ± σ):     140.863 s ±  3.346 s    [User: 160.611 s, System: 5.742 s]
  Range (min … max):   138.345 s … 148.004 s    10 runs

$ LEIN_JVM_OPTS="-agentpath:libasyncProfiler.so=start,event=alloc,total,threads,file=lein-main-1.jfr" lein-main deps :tree

# frame                         allocation size (bytes)
# clojure.main.main(String[]) - 127,910,697,208
```

Now:
```
$ hyperfine 'lein-main deps :tree'
Benchmark 1: lein-main deps :tree
  Time (mean ± σ):     47.720 s ±  0.994 s    [User: 62.377 s, System: 4.547 s]
  Range (min … max):   46.612 s … 49.555 s    10 runs

$ LEIN_JVM_OPTS="-agentpath:libasyncProfiler.so=start,event=alloc,total,threads,file=lein-main-2.jfr" lein-main deps :tree

# frame                         allocation size (bytes)
# clojure.main.main(String[]) - 40,577,168,464
```

marcobiscaro2112 committed:
Revisit cycle detection on deps graph

Related to https://codeberg.org/leiningen/leiningen/issues/29

This significantly decreases the run time (~88% faster) and memory allocations (~94% less memory allocated) for a pathological case.

Baseline (commit 96c8c90):
```
$ hyperfine 'lein-main deps :tree'
Benchmark 1: lein-main deps :tree
  Time (mean ± σ):     47.720 s ±  0.994 s    [User: 62.377 s, System: 4.547 s]
  Range (min … max):   46.612 s … 49.555 s    10 runs

$ LEIN_JVM_OPTS="-agentpath:libasyncProfiler.so=start,event=alloc,total,threads,file=lein-main-2.jfr" lein-main deps :tree

# frame                         allocation size (bytes)
# clojure.main.main(String[]) - 40,577,168,464
```

Now:
```
$ hyperfine 'lein-main deps :tree'
Benchmark 1: lein-main deps :tree
  Time (mean ± σ):      5.765 s ±  0.103 s    [User: 6.209 s, System: 0.714 s]
  Range (min … max):    5.559 s …  5.898 s    10 runs

$ LEIN_JVM_OPTS="-agentpath:libasyncProfiler.so=start,event=alloc,total,threads,file=lein-main-3.jfr" lein-main deps :tree

# frame                         allocation size (bytes)
# clojure.main.main(String[]) - 2,303,972,592
```

ChrisLane committed:
Update README URLS to link to lein scripts on Codeberg

technomancy committed:
Changelog updates.

technomancy committed:
Add static-classpath task.

technomancy committed:
Classpath exceptions shouldn't show a whole stack trace.

Catch all exceptions here, not just DependencyResolutionExceptions.

technomancy committed:
Additional protections in static-classpath.

technomancy committed:
Merge default profiles into static-classpath.

technomancy committed:
Don't read user profiles in static-classpath.

technomancy committed:
Use fennel shebang for wiki script instead of lua.

The site got migrated to a new host that doesn't have system-level lua
so we can use a manual fennel binary install instead.

$ curl -O ~/fennel https://fennel-lang.org/downloads/fennel-1.4.0-x86_64
$ chmod 755 ~/fennel

technomancy committed:
Improvements to the tutorial.

technomancy committed:
Don't use ridiculous fixtures in these tests.

These can just be locals, geez.

technomancy committed:
Work around bug in DependencyNode equality when calculating pedantic.

The DependencyNodes we are given have broken equality comparisons, so
you can end up with ridiculous output like this:

    Possibly confusing dependencies found:
    [cider/cider-nrepl "0.44.0"]
     overrides
    [cider/cider-nrepl "0.44.0"]

Make comparisons based on the Artifact from the nodes, not the nodes
themselves. We don't need to convert to a map first for this.

technomancy committed:
Use an artifact in this test which should be in the repo already.

technomancy committed:
Next version will be 2.11.0 since there's a new task added.

technomancy committed:
Update web site a bit.

Allow zoom, specify english, clarify some install details.

technomancy committed:
Apply web updates to other pages too, not just index.

technomancy committed:
Fix a bug where check task wouldn't work on code needing AOT.

technomancy committed:
Clarify profile activation.

technomancy committed:
Update template resolution to stop caching.

technomancy committed:
Remove test-paths during jar to avoid running test code accidentally.

Fixes https://github.com/technomancy/leiningen/issues/2808

marcobiscaro2112 committed:
Fix resolve of nrepl vars

The vars being used are from the nrepl.core namespace.

Also, add explicit require to nrepl.transport, since we also use a var
from it.

Fixes https://codeberg.org/leiningen/leiningen/issues/34.

marcobiscaro2112 committed:
Fix reflection warning on lein run

- Preserve metadata across process boundary
- Add Object[] tag to Reflector argument

Fixes https://github.com/technomancy/leiningen/issues/2695

technomancy committed:
Fix a bug where :eval-in :nrepl would look in the wrong place for port.

marcobiscaro2112 committed:
AOT compile all namespaces required by leiningen.core.main

We have some namespaces that are always required, regardless of which
task is running. We were not AOT compiling a few of these namespaces,
which caused a small slowdown in startup for any task (because the
Clojure compile would need to emit bytecode at runtime for all these
namespaces).

Simply including these namespaces in the AOT vector reduces over 200ms
of execution time in simple tasks:

```
$ hyperfine 'lein version'
Benchmark 1: lein version
  Time (mean ± σ):     646.6 ms ±   4.5 ms    [User: 805.5 ms, System: 58.8 ms]
  Range (min … max):   641.3 ms … 656.0 ms    10 runs

$ hyperfine 'lein-main version'
Benchmark 1: lein-main version
  Time (mean ± σ):     413.8 ms ±   3.3 ms    [User: 461.8 ms, System: 47.0 ms]
  Range (min … max):   408.6 ms … 417.8 ms    10 runs
```

jpmonettas committed:
Fix global :exclusions in profile not applying to root :dependencies

technomancy committed:
Remove stray user.clj file.

This was only meant to be added to the sample-bad-user project, not sample.

technomancy committed:
Merge remote-tracking branch 'macro/mb/aot-config'

technomancy committed:
Merge remote-tracking branch 'mb/lein-run-reflection'

technomancy committed:
Contributing and news.

technomancy committed:
Allow LEIN_STABLE to be overridden.

technomancy committed:
Release 2.11.0

technomancy committed:
Bump to 2.11.1-SNAPSHOT.

technomancy committed:
Convert password char-array into string for deploys.

Older versions of Pomegranate were tolerant of char-arrays for
passwords, but as of 2.11.0 this now causes a class cast exception.

technomancy committed:
Add tests for reading password from System/console.

We can't actually test this since System/console doesn't work with
proxy so we have to make a stupid wrapper for it and test that
instead; thanks Java.

technomancy committed:
Reset rebound-io to allow deploy test to work in full suite.

technomancy committed:
Give pedantic test project a repository to find its deps in.

Without this, tests fail when they're run on an empty local repo.

technomancy committed:
Release 2.11.1

technomancy committed:
Bump to 2.11.2-SNAPSHOT

marcobiscaro2112 committed:
Fix broken plugins

Some plugins include code with metadata that can't be read back by the reader when eval'ed.

Solves https://github.com/technomancy/leiningen/issues/2814

Reverts 707f86f24045cac3d1d4362faf332d948b7727fd, and adds a couple notes about it, because it's the second time we have the same bug trying to solve the reflection warning.

marcobiscaro2112 committed:
Link original reflection warning issue

marcobiscaro2112 committed:
Offer opt-in to include metadata in code to be evaluated in project

Add a new project key :preserve-eval-meta, to determine if metadata should be included in the code that will be evaluated in project.

Enabling that flag prevents reflection warnings when running `lein run` in a project with `:global-vars {*warn-on-reflection* true}`, but it may break plugins that spit code including objects with metadata that can't be read.

Fixes https://github.com/technomancy/leiningen/issues/2814.
Fixes https://github.com/technomancy/leiningen/issues/2695.

technomancy committed:
Merge remote-tracking branch 'marce/mb/dont-print-meta'

technomancy committed:
Document :preserve-eval-meta.

technomancy committed:
Update changelog.

technomancy committed:
Release 2.11.2

technomancy committed:
Bump to 2.11.3-SNAPSHOT

Commit:
Add Beginner Mistakes doc.

Right now it only has one entry,
but I might not be the only person
to trip over this issue.

technomancy committed:
Describe maven-based bootstrap in contributing guide.

Commit:
Fix typo in sample.project.clj

Commit:
Merge pull request 'Fix typo in sample.project.clj' (#47) from robin_lilli/leiningen:robin_lilli-patch-1 into main

Reviewed-on: https://codeberg.org/leiningen/leiningen/pulls/47

technomancy committed:
Bump pomegranate version to fix scope bug.

https://github.com/clj-commons/pomegranate/issues/125

technomancy committed:
Update classpath test for new exception class for TLS failure.

technomancy committed:
Pull in a fixed parsley version in the repl task to avoid warnings.

Fixes https://github.com/technomancy/leiningen/issues/2820

technomancy committed:
Looks like CI config has a new required field now.

mjholub committed:
chore: change non-local links to https

technomancy committed:
Use warn to emit min-lein-version warning instead of info.

Also prefix it with semicolons to avoid interfering with data.

technomancy committed:
Use semicolons before all warnings.

Just helps avoid interfering with data in cases where the output is parsed.

technomancy committed:
Stop using <tt> elements and name attributes in HTML pages.

technomancy committed:
Stop overriding Maven wagon timeout setting.

Also update the tutorial that mentions this.

The Japanese translation of the tutorial still has the incorrect
description; we will need help updating it.

See https://codeberg.org/leiningen/leiningen/issues/46

technomancy committed:
Fix test that was broken by unrelated https url change.

technomancy committed:
Fix missing delimiters in woodpecker config.

Commit:
Update nREPL version to 1.3

Commit:
Merge pull request 'Update nREPL version to 1.3' (#53) from oyakushev/leiningen:nrepl-1.3 into main

Reviewed-on: https://codeberg.org/leiningen/leiningen/pulls/53

rlbdv committed:
Ensure :active-profiles respects unmerging, say via lein pom/java

Although a project's :active-profiles metadata is not currently part
of the publically specified API, it can be useful in middlewares.

Previously init-profiles, which is used by unmerge-profiles carried
over the :active-profiles from the incoming project, and since
:active-profiles is built solely via conj, this ended up not removing
profiles that were being unmerged.  To fix that, just clear the active
profiles in init-profiles before it rebuilds them.

Commit:
Merge pull request 'Ensure :active-profiles respects unmerging, say via lein pom/java' (#54) from rlb/leiningen:unmerge-active-profiles-too into main

Reviewed-on: https://codeberg.org/leiningen/leiningen/pulls/54

gordonsyme committed:
Regenerate bootclasspath-deps.clj

Nrepl was upgraded to 1.3.0 in f82f809e31ca

Commit:
Merge pull request 'Regenerate bootclasspath-deps.clj' (#55) from gordonsyme/leiningen:gordon/sync-bootclasspath-deps into main

Reviewed-on: https://codeberg.org/leiningen/leiningen/pulls/55

technomancy committed:
Etags for leiningen.org

gabrielgiussi committed:
Add plugin-tree-data command to output plugin tree dependency as edn (analogous to tree-data)

technomancy committed:
Mention :plugin-tree-data in changelog.

technomancy committed:
Improve backslash handling on cygwin.

See https://codeberg.org/leiningen/leiningen/issues/62

Commit:
Remove palletops reference due to domain expiration.

Commit:
Merge pull request 'Remove palletops reference due to domain expiration.' (#63) from andrew-alves/leiningen:remove-pallet-link into main

Reviewed-on: https://codeberg.org/leiningen/leiningen/pulls/63

technomancy commented:
Sorry about this; for some reason github did not notify me about the changes here. This is a lot better. There's still a few issues I'd like to see fixed.

```
Possibly confusing dependencies found:
[clj-kondo "2022.12.10"] -> [org.babashka/sci "0.5.35"] -> [borkdude/edamame "1.0.0"]
 overrides
[metosin/reitit "0.7.2"] -> [metosin/reitit-malli "0.7.2"] -> [metosin/malli "0.16.4"] -> [borkdude/edamame "1.4.27"]

Consider using these exclusions:
[metosin/reitit "0.7.2" :exclusions [borkdude/edamame]]

OR

Adding the next line to the head of the project.clj file's depency vector:
[borkdude/edamame "1.4.27"]


In addition to using above exclusion method, you can also add all the following lines
to the head of the depency vector of your project.clj to resolve the confusing dependencies' problem:

[borkdude/edamame "1.4.27"]
----------------------------------------
```

* Running the tests shows a bunch of reflection warnings now
* It seems to repeat the same suggestion twice with different wording
* Misspelling of dependency
* No tests

technomancy committed:
Remove workaround for insecure HTTP wagon.

This is redundant now, pomegranate already protects us from HTTP
downloads and will give a decent error when you try to use one.

cucu committed:
wMerge branch 'master' into upgrade-to-newest-deps

cucu committed:
Merge branch 'master' into upgrade-to-newest-deps

cucu committed:
Fix
